packages/onboarding/src/HotspotOnboardingUtil.ts

```diff
@@ -0,0 +1,440 @@
+import {
+  BONES_IN_HNT,
+  DcProgram,
+  DeviceType,
+  HemProgram,
+  Maker,
+  NetworkType,
+  TXN_FEE_IN_LAMPORTS,
+} from './types'
+
+type Account = AccountInfo<string[]>
+
+const lowerFirst = (str: string) => str.charAt(0).toLowerCase() + str.slice(1)
+
+export const deviceTypeToNetworkType = (deviceType: DeviceType): NetworkType => {
+  if (deviceType === null) return 'IOT'
+
+  return 'MOBILE'
+}
+
+const hotspotInfoToDetails = (value: {
+  asset: PublicKey
+  bumpSeed: number
+  isFullHotspot?: boolean
+  location: BN | null
+  numLocationAsserts?: number
+  elevation: number | null
+  gain: number | null
+}) => {
+  const location = value.location?.toString('hex')
+  return {
+    elevation: value.elevation || undefined,
+    gain: value.gain || undefined,
+    location,
+    isFullHotspot: value.isFullHotspot,
+    numLocationAsserts: value.numLocationAsserts,
+  }
+}
+
+export const getHotspotNetworkDetails = async ({
+  address,
+  hemProgram,
+  ...opts
+}:
+  | {
+      address: string
+      networkType: NetworkType
+      hemProgram: HemProgram
+    }
+  | {
+      address: string
+      deviceType: DeviceType
+      hemProgram: HemProgram
+    }): Promise<
+  | {
+      elevation?: number
+      gain?: number
+      location?: string
+      isFullHotspot?: boolean
+      numLocationAsserts?: number
+      locationStakingFee?: BN
+    }
+  | undefined
+> => {
+  const types = { networkType: undefined, deviceType: undefined, ...opts }
+  let networkType = types.networkType
+  const deviceType = types.deviceType
+  if (!networkType && deviceType !== undefined) {
+    networkType = deviceTypeToNetworkType(deviceType)
+  }
+  if (!networkType) {
+    throw new Error('Could not determine network type')
+  }
+
+  try {
+    const mint = networkType === 'IOT' ? IOT_MINT : MOBILE_MINT
+    const subDao = subDaoKey(mint)[0]
+
+    const configKey = rewardableEntityConfigKey(subDao, networkType)
+
+    const entityConfig = await hemProgram.account.rewardableEntityConfigV0.fetchNullable(
+      configKey[0],
+    )
+    if (!entityConfig) return
+
+    if (networkType === 'IOT') {
+      const [info] = iotInfoKey(configKey[0], address)
+      const iotInfo = await hemProgram.account.iotHotspotInfoV0.fetch(info)
+
+      // @ts-ignore
+      const details = hotspotInfoToDetails(iotInfo)
+      const iotConfig = entityConfig.settings.iotConfig
+
+      const locationStakingFee = details.isFullHotspot
+        ? iotConfig?.fullLocationStakingFee
+        : iotConfig?.dataonlyLocationStakingFee
+
+      return { ...details, locationStakingFee }
+    }
+
+    const [info] = mobileInfoKey(configKey[0], address)
+
+    const mobileInfo = await hemProgram.account.mobileHotspotInfoV0.fetch(info)
+    const deviceTypeKeys = Object.keys(mobileInfo.deviceType)
+    let deviceType = ''
+    if (deviceTypeKeys.length) {
+      deviceType = deviceTypeKeys[0]
+    }
+    const mobileConfigV1 = entityConfig.settings.mobileConfigV1
```

Comment:
HIP-96 is forcing us toadd a mobileConfigV2... https://github.com/helium/helium-program-library/pull/506. Can you add support for this if present?